public/app/plugins/panel/graph/module.ts

```diff
@@ -238,7 +238,7 @@ class GraphCtrl extends MetricsPanelCtrl {
     for (const series of this.seriesList) {
       series.applySeriesOverrides(this.panel.seriesOverrides);
 
-      if (series.unit) {
+      if (this.panel.yaxes[series.yaxis - 1].format === 'none' && series.unit) {
```

Comment:
Problem needs to be fixed [here](https://github.com/grafana/grafana/blob/master/public/app/plugins/datasource/cloudwatch/datasource.ts#L140) and [here](https://github.com/grafana/grafana/blob/master/public/app/plugins/datasource/stackdriver/datasource.ts#L110) instead.

We'll create a new issue for resolving issue with that a unit cannot be changed when datasource returns a proper unit.

Reply:
Let me confirm.
If stackdriver or cloudwatch respond 'none' unit, the datasource omit unit field?
If so, I'll follow how stackdriver data source fix it, and then I'll make a PR for cloudwatch datasource.

Reply:
Sorry if I was a bit too vague. 

> If stackdriver or cloudwatch respond 'none' unit, the datasource omit unit field?

Yes exactly.

> If so, I'll follow how stackdriver data source fix it, and then I'll make a PR for cloudwatch datasource.

Stackdriver also does it wrong why I suggest to fix both cloudwatch and stackdriver in this PR. What needs to be fixed [here](https://github.com/grafana/grafana/blob/master/public/app/plugins/datasource/cloudwatch/datasource.ts#L140) and [here](https://github.com/grafana/grafana/blob/master/public/app/plugins/datasource/stackdriver/datasource.ts#L110) is that the `unit` should not be set at all in series if datasource (CloudWatch or Stackdriver) doesn't return any unit. Example solution for cloudwatch:

```diff
diff --git a/public/app/plugins/datasource/cloudwatch/datasource.ts b/public/app/plugins/datasource/cloudwatch/datasource.ts
index 7821c5d..b4f739f 100644
--- a/public/app/plugins/datasource/cloudwatch/datasource.ts
+++ b/public/app/plugins/datasource/cloudwatch/datasource.ts
@@ -137,7 +137,11 @@ export default class CloudWatchDatasource {
       if (res.results) {
         _.forEach(res.results, queryRes => {
           _.forEach(queryRes.series, series => {
-            data.push({ target: series.name, datapoints: series.points, unit: queryRes.meta.unit || 'none' });
+            const s = { target: series.name, datapoints: series.points } as any;
+            if (queryRes.meta.unit) {
+              s.unit = queryRes.meta.unit;
+            }
+            data.push(s);
           });
         });
       }
```

We can help with the Stackdriver changes if you want to focus on CloudWatch.

Reply:
Thanks for fix code.

It is easy to change Stackdriver, but I'm not responsible for it.
Thus, I fix only CloudWatch datasource.